src/surge-xt/gui/SurgeGUIEditor.cpp

```diff
@@ -4733,6 +4737,73 @@ juce::PopupMenu SurgeGUIEditor::makeMidiMenu(const juce::Point<int> &where)
     return midiSubMenu;
 }
 
+#ifdef SURGE_HAS_OSC
+juce::PopupMenu SurgeGUIEditor::makeOSCMenu(const juce::Point<int> &where)
+{
+    auto oscSubMenu = juce::PopupMenu();
+    int portnum = juceEditor->processor.oscListener.portnum;
+
+    oscSubMenu.addItem(Surge::GUI::toOSCase("List all parameters"),
+            [this]() {
+                showHTML(this->parametersToHtml()); 
+            });
+
+    oscSubMenu.addSeparator();
+    oscSubMenu
+        .addItem(
+            Surge::GUI::toOSCase("Change OSC port (" + std::to_string(portnum) + ")"),
+            [this, portnum]() {
+                const auto c{std::to_string(portnum)};
+                promptForMiniEdit(
+                    c, "Enter a new value:", "OSC Port #", juce::Point<int>(10, 10),
+                    [this](const std::string &c) {
+                        int newPort = ::atoi(c.c_str());
```

Comment:
This is a little dangerous. They could enter any value and atoi can fail.

You can use std::stoi directly (no need for atoi(c.c_str()). But you'll want to catch any exception that is thrown, and either message the user that they need to provide a valid port number or leave a default value in newPort.

Also, port numbers max out at 65535 and min out at 0. So make sure you're checking for that max value. As it is right now, a user can enter any valid integer value, even if it's not a valid port number.

Reply:
Did sanity check on user input, with error reporting.